source/include/ota.h

```diff
@@ -238,12 +245,47 @@ typedef struct OtaJobDocument
  * If the callback function is called with a result of OtaJobEventFail, the OTA update
  * job has failed in some way and should be rejected.
  *
+ * When OtaJobEventStartTest is received, the load is first boot after OTA.
+ * Users can set the image state to OtaImageStateAccepted if the load is verified.
+ *
+ * When OtaJobEventProcessed is received, the event (OtaAgentEventReceivedJobDocument) sent
+ * from user has been handled. User can free the buffer when receiving this callback.
+ *
+ * When OtaJobEventSelfTestFailed is received, that means the load is failed on verification.
+ * And the device is going to reboot after this callback.
+ *
+ * When OtaJobEventParseCustomJob is received, that means OTA received an unknown job from cloud.
+ * User can parse the job by casting pData to OtaJobDocument_t, then check the pJobDocJson and
+ * pJobId in the document. User should set the result to parseErr if it's a custom job.
+ *
+ * When OtaJobEventReceivedJob is received, that means OTA has addressed the json file provided by
+ * user successfully. Let user know to handler the buffer.
+ *
+ * When OtaJobEventUpdateComplete is received, that means OTA has downloaded a full image for the
+ * file type which is different from configOTA_FIRMWARE_UPDATE_FILE_TYPE_ID.
+ *
+ * When OtaJobEventNoActiveJob is received, that means OTA has received a job document without valid
+ * job ID and job document key.
+ *
  * @param[in] eEvent An OTA update event from the OtaJobEvent_t enum.
  *
  * @param[in] pData Optional data related to the event.
+ *
+ * eEvent|Structure of pData|Variable in pData
+ * ------|-----|-----------
+ * OtaJobEventActivate|OtaJobDocument_t|status and reason
+ * OtaJobEventFail|OtaJobDocument_t|status, reason and subReason
+ * OtaJobEventStartTest|NULL|nothing
+ * OtaJobEventProcessed|OtaEventData_t|data buffer inputed from user by OTA_SignalEvent
+ * OtaJobEventSelfTestFailed|NULL|nothing
+ * OtaJobEventParseCustomJob|OtaJobDocument_t|pJobId, jobIdLength, pJobDocJson, and jobDocLength
+ * OtaJobEventReceivedJob|OtaJobDocument_t|pJobId, jobIdLength, pJobDocJson, jobDocLength, and fileTypeId
+ * OtaJobEventUpdateComplete|OtaJobDocument_t|status, reason and subReason
+ * OtaJobEventNoActiveJob|NULL|nothing
+ *
  */
 typedef void (* OtaAppCallback_t)( OtaJobEvent_t eEvent,
-                                   const void * pData );
+                                   void * pData );
```

Comment:
Would this cause an incompatibility with the applications using ota library as in it now generates a warning of type incompatibility while assignment.

Should we also update all the demo applications with the change?

Reply:
Yes, this change would cause compile warning for the data structure. And we need to update all the demo applications with the change when we merge it.